src/state/sagas/datasets/UpdateDataset/UpdateDataset.js

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { takeEvery, call, put, select } from 'redux-saga/effects';
+import { t } from 'i18next';
+import { Api } from '../../../../services/config/Api';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
+import { DATASET_ACTIONS_KEY } from '../../../commons/DatasetConstants';
+
+const getSelectedDatasetIndex = (state) => state.dataset.selectedDatasetIndex.data;
+
+export function* updateDataset(action) {
+  const organizationId = action.organizationId;
+  const datasetId = action.datasetId;
+  const datasetData = action.datasetData;
+
+  const datasetIndex = yield select(getSelectedDatasetIndex);
```

Comment:
We don't need to get the index here, the reducer already implements the logic to retrieve the index based on the dataset id. In this sage, we can simply assume that `const datasetIndex  = action.datasetIndex;` (we don't need to create this const btw, it will only be used once)